docs/architecture/blazor-for-web-forms-developers/introduction.md

```diff
@@ -8,6 +8,50 @@ ms.date: 09/11/2019
+
+Most modern web frameworks are now also open source, which has a number of benefits. Users are not beholden to a single project owner to fix bugs and add features. Open source projects provide improved transparency on development progress and upcoming changes. Open source projects enjoy contributions from an entire community and they foster a supportive open source ecosystem. While there are still risks associated with open source, many open source consumers and contributors have found suitable mitigations (contributor license agreements, friendly licenses, pedigree scans, supporting foundations, etc.) enable them to enjoy the benefits of an open source ecosystem in a safe and reasonable way.
+
+The .NET community has embraced both cross-platform support and open source. .NET Core is an open source and cross-platform version of .NET that runs on a variety of platforms, including Windows, macOS, and different Linux distributions. Xamarin provides an open source version of .NET that runs on Android, iOS, and a variety of other form factors including watches and smart TVs. Microsoft has announced that for [.NET 5](https://devblogs.microsoft.com/dotnet/introducing-net-5/) both .NET Core and Xamarin are to be reconciled into "a single .NET runtime and framework that can be used everywhere and that has uniform runtime behaviors and developer experiences."
+
+Will ASP.NET Web Forms benefit from this move to open source and cross-platform support? The answer, unfortunately, is no, or at least not to the same extent as the rest of the platform. The .NET team [recently made it clear](https://devblogs.microsoft.com/dotnet/net-core-is-the-future-of-net/) that ASP.NET Web Forms will not be brought to .NET Core or .NET 5. Why is that? Efforts were made in the early days of .NET Core to bring over ASP.NET Web Forms, but the number of breaking changes required were found to be too drastic. There is perhaps also an admission here that even for Microsoft there is a limit to the number of web frameworks that it can support simultaneously. Perhaps someone in the community will take up the cause of creating an open source and cross-platform version of ASP.NET Web Forms. The [source code for ASP.NET Web Forms](https://github.com/microsoft/referencesource) has been made available publicly, although only in reference form. But, for the time being it seems ASP.NET Web Forms will remain Windows only and without an open source contribution model. If cross-platform support or open source become important for your scenarios, then you'll need to look for something new.
```

Comment:
```suggestion
Will ASP.NET Web Forms benefit from this move to open source and cross-platform support? The answer, unfortunately, is no, or at least not to the same extent as the rest of the platform. The .NET team [recently made it clear](https://devblogs.microsoft.com/dotnet/net-core-is-the-future-of-net/) that ASP.NET Web Forms will not be brought to .NET Core or .NET 5. Why is that? Efforts were made in the early days of .NET Core to bring over ASP.NET Web Forms, but the number of breaking changes required were found to be too drastic. There is perhaps also an admission here that even for Microsoft, there is a limit to the number of web frameworks that it can support simultaneously. Perhaps someone in the community will take up the cause of creating an open source and cross-platform version of ASP.NET Web Forms. The [source code for ASP.NET Web Forms](https://github.com/microsoft/referencesource) has been made available publicly, although only in reference form. But for the time being, it seems ASP.NET Web Forms will remain Windows-only and without an open source contribution model. If cross-platform support or open source become important for your scenarios, then you'll need to look for something new.
```